### ISSUE TITLE
Docker CLI can't be built with WSL

### ISSUE THREAD

**Issue by karolz-ms** [docker/compose-cli#1181]
**Steps to reproduce the issue:**
1. Clone compse-cli repository into your WSL (Windows Subsystem for Linux) environment
2. Make sure Docker Desktop is installed on your Windows host and that docker CLI is enabled for the WSL distro you are using. 
3. cd into compose-cli repo root and to `make`

**Describe the results you received:**
```shell
karolz@karolzp1:~/code/compose-cli-karolz$ make
[+] Building 10.6s (3/3) FINISHED                                                                         
 => [internal] load build definition from Dockerfile                                                 0.0s
 => => transferring dockerfile: 38B                                                                  0.0s
 => [internal] load .dockerignore                                                                    0.0s
 => => transferring context: 34B                                                                     0.0s
 => ERROR resolve image config for docker.io/docker/dockerfile:experimental                         10.4s
------
 > resolve image config for docker.io/docker/dockerfile:experimental:
------
failed to solve with frontend dockerfile.v0: failed to solve with frontend gateway.v0: rpc error: code = Unknown desc = error getting credentials - err: exec: "docker-credential-desktop.exe": executable file not found in $PATH, out: ``
make: *** [Makefile:40: cli] Error 1
```

**Describe the results you expected:**
Should build normally

**Output of `docker version`:**

```
Client: Docker Engine - Community
 Cloud integration: 1.0.7
 Version:           20.10.2
 API version:       1.41
 Go version:        go1.13.15
 Git commit:        2291f61
 Built:             Mon Dec 28 16:17:34 2020
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.2
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.13.15
  Git commit:       8891c58
  Built:            Mon Dec 28 16:15:28 2020
  OS/Arch:          linux/amd64
  Experimental:     true
 containerd:
  Version:          1.4.3
  GitCommit:        269548fa27e0089a8b8278fc4fc781d7f65a939b
 runc:
  Version:          1.0.0-rc92
  GitCommit:        ff819c7e9184c13b7c2607fe6c30ae19403a7aff
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

**Output of `docker context show`:**  
(it is the default and only context; docker CLI works)

**Output of `docker info`:**

```
Client:
 Context:    default
 Debug Mode: false
 Plugins:
  app: Docker App (Docker Inc., v0.9.1-beta3)
  buildx: Build with BuildKit (Docker Inc., v0.5.1-docker)
  scan: Docker Scan (Docker Inc., v0.5.0)

Server:
 Containers: 0
  Running: 0
  Paused: 0
  Stopped: 0
 Images: 6
 Server Version: 20.10.2
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 io.containerd.runtime.v1.linux runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 269548fa27e0089a8b8278fc4fc781d7f65a939b
 runc version: ff819c7e9184c13b7c2607fe6c30ae19403a7aff
 init version: de40ad0
 Security Options:
  seccomp
   Profile: default
 Kernel Version: 5.4.72-microsoft-standard-WSL2
 Operating System: Docker Desktop
 OSType: linux
 Architecture: x86_64
 CPUs: 12
 Total Memory: 24.71GiB
 Name: docker-desktop
 ID: ZH2J:BCYE:MVPX:JWI7:V7WC:SHXJ:76BS:DYEY:KHUN:LQOJ:HDDC:IPIU
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: true
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false

WARNING: No blkio weight support
WARNING: No blkio weight_device support
WARNING: No blkio throttle.read_bps_device support
WARNING: No blkio throttle.write_bps_device support
WARNING: No blkio throttle.read_iops_device support
WARNING: No blkio throttle.write_iops_device support
```

**Additional environment details (AWS ECS, Azure ACI, local, etc.):**
Microsoft Windows [Version 10.0.19042.746] (Windows 10 Enterprise version 20H2)
Docker Desktop for Windows 3.1.0 version 51484
go version go1.15.7 linux/amd64

**Comment by gtardif** [comment]
Hi @karolz-ms , thx for this feedback. 
We have some work in progress to fix this in Docker Desktop. 
In the meantime, a quick workaround that should help : delete the `credsStore` field from ~/.docker/config.json

**Comment by karolz-ms** [comment]
@gtardif thank you, the workaround got me unblocked!

**Comment by stephen-turner** [comment]
This is fixed in Docker Desktop versions 3.2.0 and 3.2.1. Closing the ticket now.
